src/txdb.cpp

```diff
@@ -319,13 +321,24 @@ bool CCoinsViewDB::Upgrade() {
         return true;
     }
 
-    LogPrintf("Upgrading database...\n");
+    int64_t count = 0;
+    LogPrintf("Upgrading utxo-set database...\n");
+    LogPrintf("[0%%]...");
     size_t batch_size = 1 << 24;
     CDBBatch batch(db);
+    uiInterface.SetProgressContinueAction(StartShutdown);
     while (pcursor->Valid()) {
         boost::this_thread::interruption_point();
+        if (ShutdownRequested()) {
+            break;
+        }
         std::pair<unsigned char, uint256> key;
         if (pcursor->GetKey(key) && key.first == DB_COINS) {
+            if (count++ % 256 == 0) {
+                uint32_t high = 0x100 * *key.second.begin() + *(key.second.begin() + 1);
+                uiInterface.ShowProgress(_("Upgrading UTXO database") +"\n"+ _("(press q to shutdown and continue later"), (int)(high * 100.0 / 65536.0 + 0.5));
```

Comment:
From my experience, translators hate messages with html or newlines in them, increasing the chance to make a mistake there.